rating/peersRatingHandler.go

```diff
@@ -129,10 +183,88 @@ func (prh *peersRatingHandler) getOldRating(pid core.PeerID) (int32, bool) {
 	return defaultRating, found
 }
 
-func (prh *peersRatingHandler) updateRatingIfNeeded(pid core.PeerID, updateFactor int32) {
-	defer prh.updateMetrics()
+func (prh *peersRatingHandler) markInactivePIDsForRemoval(receivedPIDs map[string]struct{}) {
+	removalTimestamp := prh.getTimeHandler().Add(prh.timeWaitingForReconnection).Unix()
+	topRatedPIDs := prh.topRatedCache.Keys()
```

Comment:
L188-L195 can be extracted in a function and used for both topRatedCache and badRatedCache

Reply:
done, thanks